### ISSUE TITLE
Condition changed to "offline" spontaneously, after being online.

### ISSUE THREAD

**Issue by FireWizard52** [ostrya/PresencePublisher#70]
Hello @ostrya,

Google has published the new release 2.6.2 in the Google Play store.
I updated the app to this version and everything works again as before.

Thank you very much for your quick response and solving this issue.

Best regards

_Originally posted by @FireWizard52 in https://github.com/ostrya/PresencePublisher/issues/69#issuecomment-2331652695_

**Comment by FireWizard52** [comment]
Hello @ostrya,

Yesterday I was very happy with your solution, but unfortunately, I found another issue, while running some tests.

I understood that the behavior of the detection principle has changed in recent updates. Am I correct?
Previously I was dependent on the publishing frequency. The higher the frequency, the quicker the information was published to the MQTT server. The drawback is of course the current consumption. I understood that the behavior has been changed in such a way that, if you connect or disconnect from the Wifi  a message online, resp offline is immediately published. Correct?

My problem:

If I disconnect from my Wifi, I receive an "offline" message instantly. If I connect to my Wifi again, I receive an "online" message, but (aprox.) 30 seconds after that an "offline" message is published and the app remains in this state..
I'm able to restore the "online" condition by rebooting my phone.

Can you have a look. All log filrs are attached.

[1725623018724.log](https://github.com/user-attachments/files/16907645/1725623018724.log)
[1725623073636.log](https://github.com/user-attachments/files/16907646/1725623073636.log)
[1725623092000.log](https://github.com/user-attachments/files/16907647/1725623092000.log)

Best regards

**Comment by ostrya** [comment]
Hi @FireWizard52 , sorry to hear that. Yes, you are right - the current logic is that regardless of the publishing frequency, as soon as the device connects to or disconnects from a Wi-Fi, a message is sent.

And from the logs, it looks like there are still issues with detecting the Wi-Fi name. Repeatedly, there is a "Wi-Fi disconnected: Jonas", directly followed by a "Wi-Fi connected: <unknown ssid>". I'll need to look into which mechanism I used before in Android 11, maybe something changed there.

In addition, there is your point about sending offline messages immediately. If situations like yours, i.e. disconnecting and re-connecting within a few seconds, happen frequently, we probably want to introduce a waiting time between detecting the disconnect and sending an offline message. I'll need to look into that as well.

**Comment by FireWizard52** [comment]
good evening @ostrya ,

Thank you for your quick response.

For your information `Jonas` is the network name.

The MQTT server (broker) is a local one and so the route will change.

The waiting time is probably a good idea, as, because of the connected/disconnected cycle and vice versa. the network topology changes. Perhaps a small waiting time will already be sufficient

However I did not disconnect and reconnect within seconds. Mostly it was more than a minute.
Perhaps test the availability of the MQTT server, before sending a message (both for connected and disconnected Wifi)
But I'm not sure that the issue I see, that de phone sends "offline" after sending "online" might be caused by the Network Topology change.

**Comment by ostrya** [comment]
I hope I got it now. There were several issues I had overlooked, but it should work now with the next release.

In any case, please keep an eye out for the restart issue, that should still work as before. If it persists, please create a new issue for that.

**Comment by FireWizard52** [comment]
Hi @ostrya,

I waited a couple of weeks, before to report the results of the latest release 2.6.3,
as I was too quick the last time.

I can confirm that the app detects the "Home" network correctly and it reports [online]
I did not see any issues with that, so "online" is detected properly.

However this is not the case, if I leave my house, e.g. to do some shopping.
In this case it continues to report online and it also keeps the SSID.
If I enter an area, where the SSID is known (friends), it will automatically detect that network and report as connectedWifi the SSID and [offline] So far, so good.

After I leave that network, it will properly display as connectedWifi "N/A", until I arrive at home, after it will detect the "Home" network and will report [online].

So the detection of the connection to my Home Network is okay, but the detection of the disconnection is not working.

Regards

**Comment by ostrya** [comment]
Sorry that it still does not work, and thanks for the update. So it looks like my changes are not working as intended. I guess I will need to go back to detecting the WiFi network in a time-based manner.

**Comment by ostrya** [comment]
Hi @FireWizard52, do you see any changes with the new version 2.6.4?

**Comment by FireWizard52** [comment]
Hello Kai,

I have monitored the app's behavior during  the last week and regularly checked its log file. I did not detect any issue's.
It connected to my Network, when it should do and it disconnected also, when I left the area with WiFi reception.
Also switching off/on the phone or disconnecting my WiFi has been detected as it should.

So it seems that you solved this issue, at least for me, and I would thank you for your efforts.
I suggest that I close this issue for now.

Regards
